optuna/trial/_trial.py

```diff
@@ -561,6 +564,12 @@ def objective(trial):
                 ``n_warmup_steps`` as the warmup mechanism.
         """
 
+        if len(self.study.directions) > 1:
+            _logger.warning(
+                "When the problem is multi-objective optimization, Trial.report does nothing."
+            )
+            return
```

Comment:
I'm not confident with the choice of `logging` here. According to the [Logging HOWTO](https://docs.python.org/3/howto/logging.html#when-to-use-logging), `warnings` is usually used for avoidable issues. IMO, users may want to fix their user code because `trial.report` and `trial.should_prune` do not work as expected. If we use `warnings.warn` with `stacklevel=2`, we can show the line of user code to be fixed as follows:

```console
% python examples/multi_objective/plot_pareto_front.py
[I 2020-12-23 10:04:50,754] A new study created in memory with name: no-name-bad6d896-1135-4c20-bf88-a0a29ffb521d
[I 2020-12-23 10:04:50,754] Multi-objective optimization is set, but no sampler is specified. The sampler is set to `optuna.samplers.RandomSampler`.
examples/multi_objective/plot_pareto_front.py:11: UserWarning: When the problem is multi-objective optimization, Trial.report does nothing.
  trial.report(0, 1)
examples/multi_objective/plot_pareto_front.py:12: UserWarning: When the problem is multi-objective optimization, Trial.should_prune always returns `False`.
  if trial.should_prune():
[I 2020-12-23 10:04:50,756] Trial 0 finished with values: [30.27784637812957, 19.176625243170825] and parameters: {'x': 1.603879106802351, 'y': 2.235404528333806}.
```

What do you think?

<details>
<summary>Example code</summary>

```python
import optuna


def objective(trial):
    # Binh and Korn function.
    x = trial.suggest_float("x", 0, 5)
    y = trial.suggest_float("y", 0, 3)

    v0 = 4 * x ** 2 + 4 * y ** 2
    v1 = (x - 5) ** 2 + (y - 5) ** 2
    trial.report(0, 1)
    if trial.should_prune():
        raise optuna.TrialPruned()

    return v0, v1

study = optuna.create_study(directions=["minimize", "minimize"])
study.optimize(objective, n_trials=1)
```
</details>

Reply:
Thanks for the comment. IMO, when the users face this warning, they should stop to use `trial.report` and `trial.should_prune` since those implementations are not provided.

Reply:
> they should stop to use trial.report and trial.should_prune since those implementations are not provided.

Then, I think we should use `warnings.warn` instead of `logging.warning`. Please refer to the [Logging HOWTO](https://docs.python.org/3/howto/logging.html#when-to-use-logging).

Reply:
Thanks for the suggestion. I fixed to use `warnings.warn`.